codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -102,6 +105,7 @@ public void close() { }
     @Override
     public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf data,
             int padding, boolean endStream, ChannelPromise promise) {
+        SimpleChannelPromiseAggregator promiseAggregator = null;
```

Comment:
The constructor of `SimpleChannelPromiseAggregator` may throw.  No matter what is throwing I want to ensure we are cleaning up everything. If `promiseAggregator` is created then cleanup must go though it, otherwise the `promise` is directly used.  I can this about it a bit more but it seems like I will have `promiseAggregator` defined outside a `try {..}` scope to use it in the catch block.  Any suggestions?